go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/DataDog/datadog-agent
 
-go 1.16
+go 1.17
```

Comment:
Needs a `inv -e tidy-all` I think

Reply:
I _hope_ we only need to do this once

Reply:
The "for other options" link doesn't have anything about pruning or the `-compat` flag, so I'm sort of guessing.  But, it seems like being compat with 1.17 is OK :)